src/scm/plams/tools/kftools.py

```
@@ -373,6 +398,97 @@ def read(self, section: str, variable: str, return_as_list: bool = False) -> TRe
             ret = [ret]
         return ret
 
+    def read_int(self, section: str, variable: str) -> int:
+        """Obtain an integer for a *variable* located in *section*.
+
+        :param section: Section where the variable is located
+        :param variable: Variable to obtain
+        :raises KFTypedReadError: Error is raised if the *variable* in *section* does not contain a single integer
+        :return: The integer value of *variable* located in *section*
+        """
+        ret = self.read(section, variable)
+        if not isinstance(ret, int):
+            raise KFTypedReadError(section, variable, ret, "int")
+        return ret
+
+    def read_ints(self, section: str, variable: str) -> List[int]:
+        """Obtain a list of integers for a *variable* located in *section*.
+
+        :param section: Section where the variable is located
+        :param variable: Variable to obtain
+        :raises KFTypedReadError: Error is raised if the *variable* in *section* does not contain a list of integers
+        :return: The list of integers of *variable* located in *section*
+        """
+        ret = self.read(section, variable, return_as_list=True)
+        if not isinstance(ret, List) or len(ret) == 0 or not isinstance(ret[0], int):
```

Just checking we want to raise an error if this is an empty list? There's no situation where some supposed integer list on the KF could be empty?

---

You are allowed to have an empty list, I will add an explicit read for the type in case of an empty list.